api/src/main/java/org/openmrs/module/shr/odd/generator/section/impl/ProblemSectionGenerator.java

```diff
@@ -119,6 +146,32 @@ else if(this.getSectionObs().size() > 0) // unstructured?
 		return retVal;
 	}
 
+	private Obs findLastProblemObs(Condition condition){
```

Comment:
Isn't this code copy paste? Don't repeat yourself - move this code somewhere so that it can be reused in both places. 

P.S. comments from before about using other search methods apply